src/libAtomVM/nifs.c

```diff
@@ -1284,12 +1284,46 @@ static term nif_erlang_spawn(Context *ctx, int argc, term argv[])
 static term nif_erlang_send_2(Context *ctx, int argc, term argv[])
 {
     UNUSED(argc);
+    term dest = argv[0];
+    GlobalContext *glb = ctx->global;
+    struct ListHead *processes_table_list = synclist_rdlock(&glb->processes_table);
+
+    Context *p = NULL;
+    struct ListHead *item;
+    if (term_is_pid(dest) == 1) {
+        int32_t dest_pid = term_to_local_process_id(dest);
+
+        globalcontext_send_message_nolock(glb, dest_pid, argv[1]);
+        synclist_unlock(&glb->processes_table);
+
+    } else if (term_is_atom(dest) == 1) {
+        int atom_index = term_to_atom_index(dest);
+
+        int32_t dest_pid = -1;
+        int32_t test_pid = globalcontext_get_registered_process(glb, atom_index);
+        if (UNLIKELY(test_pid == 0)) {
+            synclist_unlock(&glb->processes_table);
+            RAISE_ERROR(BADARG_ATOM);
+        }
 
-    term pid_term = argv[0];
-    VALIDATE_VALUE(pid_term, term_is_pid);
+        LIST_FOR_EACH (item, processes_table_list) {
+            p = GET_LIST_ENTRY(item, Context, processes_table_head);
+            if (p->process_id == test_pid) {
+                dest_pid = test_pid;
+            }
+        }
```

Comment:
You may want to replace this loop with a call to `globalcontext_get_process_nolock`.

Reply:
That is declared as static, but if you don't mind me exposing it, that would be cleaner. That was actually what I tried using at first, but resorted to using the same loop as `globalcontext_get_process_nolock` when I found it wasn't exposed.